composer.json

```
@@ -11,12 +11,13 @@
     }
   ],
   "require": {
-    "guzzlehttp/guzzle": "^7.0",
-    "php": "~8.1.0 || ~8.2.0 || ~8.3.0"
+    "guzzlehttp/guzzle": "^7.9.2",
+    "guzzlehttp/promises": "^2.0.4",
+    "php": "~8.1.0 || ~8.2.0 || ~8.3.0 || ~8.4.0"
   },
   "require-dev": {
-    "phpunit/phpunit": "^10.5",
-    "ergebnis/phpunit-slow-test-detector": "^2.15"
+    "phpunit/phpunit": "^10.5.45 || ^11.5 || ^12.0",
```

Supporting multiple phpunit versions can be complex, and I don't think is necessary. I think we just need to bump the minimum to a version of `10.5` that works on PHP8.4, so that it runs happily in the `--lowest` build.

```suggestion
    "phpunit/phpunit": "^10.5.45",
```